borg/repository.py

```diff
@@ -713,7 +779,7 @@ def write_delete(self, id, raise_full=False):
         crc = self.crc_fmt.pack(crc32(id, crc32(header)) & 0xffffffff)
         fd.write(b''.join((crc, header, id)))
         self.offset += self.put_header_fmt.size
-        return self.segment
+        return self.segment, self.put_header_fmt.size
```

Comment:
write_put returns (segment, offset)

write_delete used to return segment, now (segment, length)

write_commit returns None

Not really consistent either way.